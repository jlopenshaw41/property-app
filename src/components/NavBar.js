import React from "react";
import "../styles/NavBar.css";
import logo from "../images/logo.png";

const NavBar = () => {
  return (
    <div className="navbar">
      <img src={logo} alt="Logo" data-testid="logo" />
      <div className="navbar-links">
        <ul>
          <li className="navbar-links-item" data-testid="nav-link-1">
            <a href="www.google.com">View properties</a>
          </li>
          <li className="navbar-links-item" data-testid="nav-link-2">
            <a href="www.google.com">Add a property</a>
          </li>
        </ul>
      </div>
    </div>
  );
};

export default NavBar;
