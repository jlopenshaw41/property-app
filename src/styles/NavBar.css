.navbar {
    background-color: white;
    border-bottom: 1px solid gray;
    display: flex;
    justify-content: space-between;
    align-items: center;
}

img {
    height: 50px;
}

.navbar-links {
    margin: 0;
    padding: 0;
}

.navbar-links-item {
    margin: 0;
    padding: 10px;
    display: inline-block;
}