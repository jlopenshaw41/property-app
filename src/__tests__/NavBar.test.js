import React from "react";
import { render } from "@testing-library/react";
import NavBar from "../components/NavBar";

describe("NavBar", () => {
  it("renders as expected", () => {
    const { asFragment } = render(<NavBar />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders logo", () => {
    const { getByTestId } = render(<NavBar />);
    const logo = getByTestId("logo");
    expect(logo).toBeTruthy();
  });

  it("renders nav links", () => {
    const { getByTestId } = render(<NavBar />);
    const navLink1 = getByTestId("nav-link-1");
    const navLink2 = getByTestId("nav-link-2");
    expect(navLink1).toBeTruthy();
    expect(navLink2).toBeTruthy();
  });
});
