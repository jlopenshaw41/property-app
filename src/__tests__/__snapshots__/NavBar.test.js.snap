// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders as expected 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <img
      alt="Logo"
      data-testid="logo"
      src="logo.png"
    />
    <div
      class="navbar-links"
    >
      <ul>
        <li
          class="navbar-links-item"
          data-testid="nav-link-1"
        >
          <a
            href="www.google.com"
          >
            View properties
          </a>
        </li>
        <li
          class="navbar-links-item"
          data-testid="nav-link-2"
        >
          <a
            href="www.google.com"
          >
            Add a property
          </a>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
